### PR TITLE
updated getProfile method using latest instance of the token for PDS …

### PR DESCRIPTION
**Pull request by KaranP25**
…member

## Proposed changes

Issue was found in 'packages/zowe-explorer/src/dataset/ZoweDatasetNode.ts' in the openDS() method where it tries to fetch contents if local copy is not found... The code this.getProfile() is acquiring the bearer token where the token does not get refreshed after the expire time. To, fix it I saw in USS we are using "Profiles.getInstance().loadNamedProfile(this.getProfileName())" to call the profile and this does return the profile with the updated token.
https://github.com/zowe/zowe-explorer-vscode/issues/3001

## Release Notes

Milestone:

Changelog:

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (non-breaking change which adds or improves functionality)
- [ ] Breaking change (a change that would cause existing functionality to not work as expected)
- [ ] Documentation (Markdown, README updates)
- [ ] Other (please specify above in "Proposed changes" section)

## Checklist

**General**

- [ ] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/zowe-explorer-vscode/wiki/Best-Practices:-Contributor-Guidance) wiki
- [ ] All PR dependencies have been merged and published (if applicable)
- [ ] A GIF or screenshot is included in the PR for visual changes
- [ ] The pre-publish command has been executed:
  - **v2 and below:** `yarn workspace vscode-extension-for-zowe vscode:prepublish`
  - **v3:** `pnpm --filter vscode-extension-for-zowe vscode:prepublish`

**Code coverage**

- [ ] There is coverage for the code that I have added
- [ ] I have added new test cases and they are passing
- [ ] I have manually tested the changes

**Deployment**

- [ ] I have added developer documentation (if applicable)
- [ ] Documentation should be added to Zowe Docs
  - If you're an outside contributor, please post in the [#zowe-doc Slack channel](https://openmainframeproject.slack.com/archives/CC961JYMQ) to coordinate documentation.
  - Otherwise, please check with the rest of the squad about any needed documentation before merging.
- [ ] These changes may need ported to the appropriate branches (list here):

### CODE REVIEW

**Comment by KaranP25** [comment]
Issue #3001

**Comment by JillieBeanSim** [comment]
Hey @KaranP25, thanks for this contribution.  We will begin testing the functionality changes, would you mind addressing any failing checks? Thanks 😸

File: packages/zowe-explorer/src/dataset/ZoweDatasetNode.ts
Comment (zFernand0) [inline packages/zowe-explorer/src/dataset/ZoweDatasetNode.ts:43]: if possible, please import the instance of imperative provided by the `@zowe/zowe-explorer-api` package.

File: packages/zowe-explorer/src/dataset/ZoweDatasetNode.ts
Comment (zFernand0) [inline packages/zowe-explorer/src/dataset/ZoweDatasetNode.ts:459]: Should we have a little trace-log here?
`ZoweLogger.trace("ZoweDatasetNode.getProfile called.");`

**Review by zFernand0 — CHANGES_REQUESTED** [review]
Left a small comment which hopefully doesn't cause too much trouble 😋 
will be testing the functionality in a few moments, but changes LGTM! 😋

File: packages/zowe-explorer/src/dataset/ZoweDatasetNode.ts
Comment (traeok) [inline packages/zowe-explorer/src/dataset/ZoweDatasetNode.ts:461]: While its highly unlikely, a data set node can technically be created without a profile, so `prof` may be `undefined` for certain nodes. In addition, the `name` property on a profile is optional. Can you please add a check to see if `prof.name` exists before passing it to `Profiles.loadNamedProfile`? This would also avoid any warnings or flags by security scanning tools for the use of `prof.name`.

```suggestion
        // return the latest object for the given profile name
        return prof.name ? Profiles.getInstance().loadNamedProfile(prof.name) : undefined;
```

**Review by traeok — COMMENTED** [review]
The fix makes sense to me, thanks @KaranP25! I left a comment regarding the use of `prof.name` - will approve once addressed.

**Comment by KaranP25** [comment]
Hi, thank you for your suggestions I have added them to my change set. Test seems to pass as well now

File: packages/zowe-explorer/src/dataset/ZoweDatasetNode.ts
Comment (zFernand0) [inline packages/zowe-explorer/src/dataset/ZoweDatasetNode.ts:462]: Hey @KaranP25,
Was this commented out by mistake? 😋

Comment (KaranP25) [reply packages/zowe-explorer/src/dataset/ZoweDatasetNode.ts:462]: Oops shouldn't be like that, let me take a look

Comment (KaranP25) [reply packages/zowe-explorer/src/dataset/ZoweDatasetNode.ts]: I have updated this to include the method

**Comment by KaranP25** [comment]
I have updated the code as the getProfile will work for Datasets and the failing test cases should be fixed now

File: packages/zowe-explorer/src/dataset/ZoweDatasetNode.ts
Comment (traeok) [inline packages/zowe-explorer/src/dataset/ZoweDatasetNode.ts:462]: I noticed that the function implementation was renamed to `getProfileInstance`. Is this change intentional? We currently use the `getProfile` function in several other places in the context of data set nodes - I think it would be best to implement the fix within the `getProfile` implementation itself so that the fix is applied everywhere. Otherwise, we'll need to update some references in the `dataset/actions.ts` file so that it refers to the proper profile.

Comment (KaranP25) [reply packages/zowe-explorer/src/dataset/ZoweDatasetNode.ts:462]: The original fix was causing a bunch of test issues and since this fix was relatively for opening of datasets and getting the children, I decided to change up the name and make it specific to just this class but still call the getProfile name. But, the original getProfile is still there in packages/zowe-explorer-api/src/tree/ZoweTreeNode.ts, so other classes shouldn't be affected. At first I was unable to import the "Profiles" to the original getProfile method, otherwise I think we would see less test failures.

Comment (KaranP25) [reply packages/zowe-explorer/src/dataset/ZoweDatasetNode.ts]: After discussing with Billie, we will stick to the getProfile Name and I will work on the failing test cases.

**Comment by KaranP25** [comment]
I have fixed the failing test cases and updated the changelog. Thank you!

File: packages/zowe-explorer/__tests__/__unit__/job/actions.unit.test.ts
Comment (JillieBeanSim) [inline packages/zowe-explorer/__tests__/__unit__/job/actions.unit.test.ts]: this commented code can be removed?

Comment (JillieBeanSim) [reply packages/zowe-explorer/__tests__/__unit__/job/actions.unit.test.ts]: I can push that to not hold up other reviews

**Review by JillieBeanSim — APPROVED** [review]

**Review by t1m0thyj — COMMENTED** [review]
LGTM, thanks @KaranP25! Just one comment about the changelog

**Review by JillieBeanSim — APPROVED** [review]

**Review by traeok — APPROVED** [review]
LGTM, thanks @KaranP25!

**Review by zFernand0 — APPROVED** [review]

**Pull request merged**
